cms/djangoapps/contentstore/courseware_index.py

```diff
@@ -45,12 +115,103 @@ class CoursewareSearchIndexer(object):
+        AboutInfo("course_staff_extended", AboutInfo.ANALYSE, AboutInfo.fetch_from_about),
+        AboutInfo("requirements", AboutInfo.ANALYSE, AboutInfo.fetch_from_about),
+        AboutInfo("syllabus", AboutInfo.ANALYSE, AboutInfo.fetch_from_about),
+        AboutInfo("textbook", AboutInfo.ANALYSE, AboutInfo.fetch_from_about),
+        AboutInfo("faq", AboutInfo.ANALYSE, AboutInfo.fetch_from_about),
+        AboutInfo("more_info", AboutInfo.ANALYSE, AboutInfo.fetch_from_about),
+        AboutInfo("ocw_links", AboutInfo.ANALYSE, AboutInfo.fetch_from_about),
+        AboutInfo("enrollment_start", AboutInfo.PROPERTY, AboutInfo.fetch_course_property),
+        AboutInfo("enrollment_end", AboutInfo.PROPERTY, AboutInfo.fetch_course_property),
+        AboutInfo("org", AboutInfo.PROPERTY, AboutInfo.fetch_course_property),
+    ]
+
+    @classmethod
+    def index_about_information(cls, modulestore, course):
+        """
+        Add the given course to the course discovery index
+
+        Arguments:
+        modulestore - modulestore object to use for operations
+
+        course - course object from which to take properties, locate about information
+        """
+        searcher = SearchEngine.get_search_engine(INDEX_NAME)
+        if not searcher:
+            return
+
+        course_id = unicode(course.id)
+        course_info = {
+            'id': course_id,
+            'course': course_id,
+            'content': {},
+            'image_url': course_image_url(course),
+        }
+
+        for about_information in cls.ABOUT_INFORMATION_TO_INCLUDE:
+            # Broad exception handler so that a single bad property does not scupper the collection of others
+            try:
+                section_content = about_information.get_value(course, modulestore)
+            except Exception as err:  # pylint: disable=broad-except
+                section_content = None
+                log.warning(
+                    "Course discovery could not collect property %s for course %s - %r",
+                    about_information.property_name,
+                    course_id,
+                    err,
+                )
+
+            if section_content:
+                if about_information.index_flags & AboutInfo.ANALYSE:
+                    analyse_content = section_content
+                    if isinstance(section_content, basestring):
+                        analyse_content = strip_html_content_to_text(section_content)
+                    course_info['content'][about_information.property_name] = analyse_content
+                if about_information.index_flags & AboutInfo.PROPERTY:
+                    course_info[about_information.property_name] = section_content
+
+        # Broad exception handler to protect around and report problems with indexing
+        try:
+            searcher.index(DISCOVERY_DOCUMENT_TYPE, course_info)
+        except Exception as err:  # pylint: disable=broad-except
```

Comment:
minor / optional:  this can be done a little more succinctly e.g.
```
except:
    log.error("something went wrong with %s", course_id, exc_info=True)
    raise
```
this also ensures you'll get the full content of the stacktrace in the log, no matter what else might catch/handle the exception later.